src/dataframe/dataframe.jl

```diff
@@ -368,7 +380,7 @@ end
 function Base.setindex!(df::DataFrame,
                 v::Any,
                 col_ind::ColumnIndex)
-    insert_single_column!(df, upgrade_scalar(df, v), col_ind)
+    fill_single_column!(df, v, col_ind)
```

Comment:
Sorry. What I meant was to use `setindex` for the `DataVector`/`NullableVector` which would probably be something like
```df
df[:col_ind][:] = v
```